tests/lib/mirrors/reflected_type_generics_test.dart

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2013, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library test.reflected_type_generics_test;
+
+import 'dart:mirrors';
+
+import 'package:expect/expect.dart';
+
+import 'reflected_type_helper.dart';
+
+class A<T> {}
+class G {}
+
+main() {
+  expectReflectedType(reflectType(A, [G]), new A<G>().runtimeType);
```

Comment:
Add tests for
 - providing empty list of type arguments for a non-generic type.
 - providing some type arguments for a non-generic type
 - providing too many / too few type argument for a generic type
 - a non-type in the type arguments list

And cases where the VM's internal representation of the type have a different number of type argument that the language model:

```
class Super<T> { }
class Sub extends Super<int> {}
reflectType(Sub, [String]) // error
```

```
class Super<T> {}
class Sub<S> extends Super<S> {}
reflectType(Sub, [String]) // okay
reflectType(Sub, [String, String]) // error
```